### DISCUSSION TITLE
TabView and awaiting lazyView.LoadViewAsync

### DISCUSSION THREAD

**Discussion by Jared-Aus** [xamarin/XamarinCommunityToolkit#1204] [Q&A]
The current marriage between TabView and LazyView seems somewhat broken to me. If the only requirement was for the initial tab or page to be not slowed down by the loading of other tab items, a simple IsLazyLoaded property might have been sufficient. By utilising LazyView and having the TabView implementation tied to BaseLazyView there is seemingly more power for developers, however that power is immediately taken away as there's no control over when the lazy views are loaded. If your second tab item takes 3 seconds to load and you want to use a custom LazyView implementation that starts life as an ActivityIndicator, the tab doesn't switch for 3 seconds as it's waiting for the lazy view to fully load. My guess is that's a bug in the implementation, but it could be intentional for all I know.
Wrapping the lazy view in a ContentView or nesting lazy views gets around the tight coupling, but then we're back to square one. A basic feature would ideally be simpler to use, or a more powerful one less rigid.

**Comment by pictos** [comment]
@Jared-Aus the `TabView` uses the `BaseLazyView` as a reference and you can extend that if you need it. Yes, the XCT `LazyView` is very simple implementation and we know that. IF you see the XamarinShow about `LazyView`, you will see that I showed how you can extend the LazyView and customize how the view will load and how it will appear.

And as I said on the show, if you want another kind of LazyView, maybe one that supports an AcitivityIndicator, open an issue as a feature request and let's talk about it.

**Reply by Jared-Aus** [reply]
I think my issue is more with the TabView than LazyView. I have extended BaseLazyView as you demonstrated in the video and it works well when disconnected from TabView's automatic LoadViewAsync calls. My issue is the two halves of your presentation only work in isolation. Have you seen what happens when you use your extended LazyView as a second or third tab item?

**Reply by pictos** [reply]
TBH no, can you create an issue and attach a repro on it?

**Reply by Jared-Aus** [reply]
I'll hopefully find some time on the weekend for that.
